core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -0,0 +1,649 @@
+        showMessage(
+          'Exploration is set to Public and is accessible to Oppia users.'
+        );
+      } else {
+        throw new Error(
+          'Exploration is set to Private and is not' +
+            ' accessible to Oppia users.'
+        );
+      }
+    }
+  }
+
+  /**
+   * This function helps in adding voice artist.
+   */
+  async addVoiceArtists(voiceArtists: string[]): Promise<void> {
+    for (let i = 0; i < voiceArtists.length; i++) {
+      await this.clickOn(voiceArtistEditButton);
+      await this.clickOn(addVoiceArtistUserName);
+      await this.page.waitForSelector(
+        `${addVoiceArtistUserName}:not([disabled])`
+      );
+      await this.page.click(addVoiceArtistUserName, {clickCount: 3});
+      await this.page.keyboard.press('Backspace');
+      await this.type(addVoiceArtistUserName, voiceArtists[i]);
+      await this.clickOn(voiceArtistSaveButton);
+
+      showMessage(voiceArtists[i] + ' has been added successfully.');
+    }
+  }
+
+  /**
+   * This function helps to choose notification type.
+   */
+  async optInToEmailNotifications(): Promise<void> {
+    await this.clickOn(feedbackToggleOff);
+  }
+
+  /**
+   * Exception function to verify the choice of receiving feedback
+   * and suggestion notifications via email
+   */
+  async expectEmailNotificationsToBeActivated(): Promise<void> {
+    await this.page.waitForSelector('input[id="feedback-switch"]');
+    const input = await this.page.$('input[id="feedback-switch"]');
+
+    if (!input) {
+      throw new Error('Feedback switch input element not found.');
+    }
+    const isChecked = await input.evaluate(
+      input => (input as HTMLInputElement).checked
+    );
+
+    if (!isChecked) {
+      showMessage('suggestions notifications via email are enabled.');
+    } else {
+      throw new Error('suggestions notifications via email are disabled.');
+    }
+  }
+
+  /**
+   * This funciton helps in deleting the exploration successfully.
+   */
+  async deleteExploration(): Promise<void> {
+    await this.page.waitForSelector(
+      `${deleteExplorationButton}:not([disabled]`
+    );
+    await this.clickOn(deleteExplorationButton);
+    await this.clickOn(deleteConfirmButton);
+  }
+
+  /**
+   * This function helps in verifying, if exploration is
+   * deleted successfully or not.
+   */
+  async expectExplorationToBeDeletedSuccessfullyFromCreatorDashboard(): Promise<void> {
+    try {
+      await this.goto(explorationUrlAfterPublished);
+      throw new Error('Exploration is not deleted successfully.');
+    } catch (error) {
+      showMessage('Exploration is successfully deleted.');
+    }
+  }
+
+  /**
+   * This function helps in drafting the exploration.
+   */
+  async saveDraftExploration(): Promise<void> {
+    await this.page.keyboard.press('Tab');
+    if (this.isViewportAtMobileWidth()) {
+      await this.clickOn(mobileSaveDraftButton);
+    } else {
+      await this.clickOn(saveDraftButton);
```

Comment:
I dont think we need to explicitly mention `desktopSaveDraftButton`. As it is obvious if we are using it normally then  it will be for desktop. We are using `mobile` keyword for mobiles.
Thanks.

Reply:
I don't think we should think of the desktop as "normal". We want to treat desktop and mobile as equally important, so I suggest that you keep parity between both.